lib/ar_lazy_preload/active_record/relation.rb

```diff
@@ -75,6 +89,32 @@ def preloads_associations_lazily?
       @preloads_associations_lazily ||= false
     end
 
+    def handle_preloaded_records(preloaded_records)
+      return unless Contexts::TemporaryPreloadConfig.enabled? || preloads_associations_lazily?
+
+      records_array = preloaded_records_to_array(preloaded_records)
+
+      return unless records_array&.any?
```

Comment:
is it possible that `records_array` is `nil`?

Reply:
It could fail on two hypothetical scenarios 
1. if the new version of rails is going to return some new class as a result of the preloader
2. if the ActiveRecord::Relation is not preloaded for some reason (I can't imagine when it's not going to be preloaded, because we explicitly preloaded them). 

thinking about the first scenario, I think it's better to fail with some meaningful error instead of a silent return.
```
def preloaded_records_to_array(records)
  case records
  when Array
    records
  when ::ActiveRecord::Relation
    records.to_a if records.loaded?
  else 
    raise(ArgumentError, 'Unsopported class for preloaded records')
  end
end
```

About 2nd scenario, I'm not sure here, do you think it's a good safe to always call `to_a` on relation?

Reply:
Yeah, I guess we could raise an error in the else close inside `preloaded_records_to_array`. 

> do you think it's a good safe to always call to_a on relation?

I guess so, at least until we don't know the scenario when relation was not preloaded